packages/core/src/scratchorg/pool/ClientSourceTracking.ts

```diff
@@ -60,31 +60,49 @@ export default class ClientSourceTracking {
 
             const sfpowerscriptsArtifacts = await this.org.getInstalledArtifacts();
 
+            //clean up MPD to just one package, so that source tracking lib
+            //does do a full scan and break
+            this.cleanupSFDXProjectJsonTonOnePackage(tempDir.name, sfpowerscriptsArtifacts[0].Name);
+
             const project = await SfdxProject.resolve(tempDir.name);
 
             // Create local source tracking files in temp repo
             const tracking = await SourceTracking.create({
                 org: this.org,
                 project: project,
             });
-            tracking.ensureLocalTracking();
 
             git = simplegit(tempDir.name);
 
-            SFPLogger.log(`Total Artifacts to Analyze: ${sfpowerscriptsArtifacts.length}`, LoggerLevel.INFO);
+            SFPLogger.log(
+                `Total Artifacts to Analyze: ${sfpowerscriptsArtifacts.length}`,
+                LoggerLevel.INFO,
+                this.logger
+            );
 
             let count = 1;
             for (const artifact of sfpowerscriptsArtifacts) {
-                SFPLogger.log(EOL, LoggerLevel.INFO);
-                SFPLogger.log(COLOR_HEADER(`Package ${count} of ${sfpowerscriptsArtifacts.length}`), LoggerLevel.INFO);
-                SFPLogger.log(`Analyzing package ${COLOR_KEY_MESSAGE(artifact.Name)}`, LoggerLevel.INFO);
+                SFPLogger.log(EOL, LoggerLevel.INFO, this.logger);
+                SFPLogger.log(
+                    COLOR_HEADER(`Package ${count} of ${sfpowerscriptsArtifacts.length}`),
+                    LoggerLevel.INFO,
+                    this.logger
+                );
+                SFPLogger.log(`Analyzing package ${COLOR_KEY_MESSAGE(artifact.Name)}`, LoggerLevel.INFO, this.logger);
+
+                //Reset the project json before going for full track
+                await git.raw(['restore', 'sfdx-project.json']);
```

Comment:
Cleaner if we just force checkout

Reply:
I looked at the faster option ( restore vs checkout ), as we only change one file

Reply:
It's probably faster to do a force checkout than two operations: restore and checkout

Reply:
makes sense, fixing it